modin/core/io/text/text_file_dispatcher.py

```diff
@@ -963,6 +964,7 @@ def _read(cls, filepath_or_buffer, **kwargs):
         new_query_compiler : BaseQueryCompiler
             Query compiler with imported data for further processing.
         """
+        filepath_or_buffer = stringify_path(filepath_or_buffer)
```

Comment:
Why is this needed?

Reply:
@anmyachev it's a nice utility function used by pandas. Sometimes users pass in Path-like objects and we need to attempt to stringify the path. If it ends up being just a buffer, this method will handle that for us as well.

Reply:
Doesn't this method do part of the job as our functions `get_path` and `get_path_or_buffer`? In this case, we could refuse to use our functions.

Reply:
Not for get_path since that function expects a string to be passed in. Unsure about the latter

Reply:
It seems to me the code before worked with Path-like types without `stringify_path`, didn't it?

Reply:
Yes but now that I think about it, it might be better to use stringify_path in place of both get_path and get_path_or_buffer.

Reply:
@anmyachev reading the warning in `get_path_or_buffer` makes me wonder if we should leave our current implementation as is?

Reply:
Yes, `stringify_path` cannot replace `get_path_or_buffer`.

Reply:
We can try to replace `get_path` with `stringify_path` in a separate pull request if needed.

Reply:
@anmyachev yes, I reverted things back for now. Please take a look and approve if it looks good!